### PR TITLE
Added support for SQLite.

### LINKED ISSUE #25: DatabaseGPT for SQLite
DatabaseGPT currently supports scenarios where we have SqlServer or Postgre databases. It doesn't currently support the case of a different like SQLite, MySql, etc... The major difference between all these DBMS is mainly the dialect of the SQL language used in the database and the required drivers themselves.

The implementation for SQLitebe implemented in the project `DatabaseGpt.SQLite`, in a similar way to what was done for DatabaseGpt.SqlServer and DatabaseGpt.Npgsql

### PR DESCRIPTION
**Pull request by nanny07**
Fix #25

### CODE REVIEW

**Review by marcominerva — CHANGES_REQUESTED** [review]

File: src/DatabaseGpt.Sqlite/DatabaseGpt.Sqlite.csproj [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Dapper" Version="2.1.28" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="8.0.1" />
```
[the remark this answers is not in the document]
Comment (nanny07) [reply src/DatabaseGpt.Sqlite/DatabaseGpt.Sqlite.csproj:11]: You are right, I wasn't paying attention but the only package needed is "Microsoft.Data.Sqlite"

File: src/DatabaseGpt.Sqlite/SqliteDatabaseGptProvider.cs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    public string Name => "SQLite";
+
+    public string Language => "SQLite";
+
+    public async Task<IEnumerable<string>> GetTablesAsync(IEnumerable<string> includedTables, IEnumerable<string> excludedTables, CancellationToken cancellationToken = default)
+    {
+        ThrowIfDisposed();
+
+        var tables = await connection.QueryAsync<string>("""
+            SELECT TBL_NAME AS Tables
```
[the remark this answers is not in the document]
Comment (nanny07) [reply src/DatabaseGpt.Sqlite/SqliteDatabaseGptProvider.cs:26]: SQLite doesn't support SCHEMA on the table name like other databases so I've removed that part. [Rerefence](https://www.sqlite.org/lang_naming.html)

How can we handle this?

File: src/DatabaseGpt.Sqlite/SqliteDatabaseGptProvider.cs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        var result = new StringBuilder();
+
+        foreach (var table in tables)
+        {
+            var query = $"""
+                SELECT '[' || NAME || '] ' ||
+                    UPPER(TYPE) || ' ' ||
+                    CASE WHEN [NOTNULL] = 0 THEN 'NULL' ELSE 'NOT NULL' END
+                FROM PRAGMA_TABLE_INFO(@table)
+                WHERE NAME NOT IN (@excludedColumns);
```
[the remark this answers is not in the document]
Comment (nanny07) [reply src/DatabaseGpt.Sqlite/SqliteDatabaseGptProvider.cs:56]: Aren't this enought?
"WHERE NAME NOT IN (@excludedColumns)" 

image

image

[the remark this answers is not in the document]
Comment (marcominerva) [reply src/DatabaseGpt.Sqlite/SqliteDatabaseGptProvider.cs:56]: Even if you don't have to deal with Schema, there still the need to handle the 'Table.ColumnToExclude' case.

**Review by marcominerva — APPROVED** [review]

**Pull request merged**